akka-actor/src/main/scala/akka/pattern/RetrySupport.scala

```diff
@@ -16,9 +17,70 @@ import scala.util.control.NonFatal
 trait RetrySupport {
 
   /**
-   * Given a function from Unit to Future, returns an internally retrying Future
-   * The first attempt will be made immediately, each subsequent attempt will be made after 'delay'
-   * A scheduler (eg context.system.scheduler) must be provided to delay each retry
+   * Given a function from Unit to Future, returns an internally retrying Future.
+   * The first attempt will be made immediately, each subsequent attempt will be made immediately
+   * if the previous attempt failed.
+   * If attempts are exhausted the returned future is simply the result of invoking attempt.
+   * Note that the attempt function will be invoked on the given execution context for subsequent
+   * tries and therefore must be thread safe (not touch unsafe mutable state).
```

Comment:
Would be good if this was separate paragraph (empty line above it) so that it stands out a bit more